soh/src/overlays/actors/ovl_En_Daiku/z_en_daiku.c

```diff
@@ -403,6 +405,7 @@ void EnDaiku_InitEscape(EnDaiku* this, PlayState* play) {
     this->stateFlags &= ~(ENDAIKU_STATEFLAG_1 | ENDAIKU_STATEFLAG_2);
 
     gSaveContext.eventChkInf[EVENTCHKINF_CARPENTERS_FREE_INDEX] |= EVENTCHKINF_CARPENTERS_FREE_MASK(this->actor.params & 3);
+    GameInteractor_ExecuteOnFlagSet(FLAG_EVENT_CHECK_INF, (EVENTCHKINF_CARPENTERS_FREE_INDEX << 4) + (int)log2(EVENTCHKINF_CARPENTERS_FREE_MASK(this->actor.params & 3)));
```

Comment:
another instance where i think it makes sense to have an array with the values we need instead of including math and casting to int

Reply:
This one in particular can actually just become the following since `this->actor.params & 3` is already in the range of 0-3
The _MASK macro is shifting the value, and log2 is unshifting the result back
```suggestion
    GameInteractor_ExecuteOnFlagSet(FLAG_EVENT_CHECK_INF, (EVENTCHKINF_CARPENTERS_FREE_INDEX << 4) + (this->actor.params & 3));
```